src/main/scala/firrtl/Compiler.scala

```diff
@@ -27,42 +27,61 @@ object RenameMap {
 // TODO This should probably be refactored into immutable and mutable versions
 final class RenameMap private () {
   private val underlying = mutable.HashMap[Named, Seq[Named]]()
-  /** Get new names for an old name
-    *
-    * This is analogous to get on standard Scala collection Maps
-    * None indicates the key was not renamed
-    * Empty indicates the name was deleted
+
+  /** Get renames of a [[CircuitName]]
+    * @note A [[CircuitName]] can only be renamed to one-or-more [[CircuitName]]s
     */
-  // TODO Is there a better way to express this?
-  def get(key: Named): Option[Seq[Named]] = {
+  def get(key: CircuitName): Option[Seq[CircuitName]] = underlying.get(key).map {
```

Comment:
How do you rename it to multiple `CircuitName`s? Is that splitting the circuit into two circuits? Does the rest of firrtl support that?